tests/h/presenters/annotation_json_test.py

```diff
@@ -221,19 +199,24 @@ def test_permissions(
         presenter = AnnotationJSONPresenter(resource)
         assert expected == presenter.permissions[action]
 
-    def test_exception_for_wrong_formatter_type(self):
-        with pytest.raises(ValueError) as exc:
-            AnnotationJSONPresenter(mock.Mock(), formatters=[mock.Mock()])
-
-        assert "not implementing IAnnotationFormatter interface" in str(exc.value)
-
     @pytest.fixture
-    def document_asdict(self, patch):
-        return patch("h.presenters.annotation_json.DocumentJSONPresenter.asdict")
+    def get_formatter(self):
+        def get_formatter(payload=None):
+            formatter = create_autospec(
+                AnnotationFormatter, spec_set=True, instance=True
+            )
+            formatter.format.return_value = payload or {}
+            return formatter
+
+        return get_formatter
```

Comment:
I guess being able to autospec a mock from it is one advantage of the interface actually being defined in a class (or `ABC`). That could be a perfectly valid reason for using an `ABC` here rather than an informal protocol